quinn/src/lib.rs

```diff
@@ -37,6 +37,64 @@
 //! authority. If this is infeasible--for example, if servers are short-lived or not associated
 //! with a domain name--then as with TLS, self-signed certificates can be used to provide
 //! encryption alone.
+//!
+//! # Feature flags
+//!
+//! ### Essential features
+//!
+//! Usage without these features is not currently supported. TODO: If quinn can be used without rustls then
+//! the exact mechanism of that should be documented, because from the code it's unclear how rustls could
+//! be possibly pulled out.
+//!
+//! - `rustls`: Currently this must be enabled. In the future other SSL libraries may be supported.
+//!
+//! ### Miscellaneous features
+//!
+//! Miscellaneous optional features.
+//!
+//! - `bloom`: Enabled by default. Enables `BloomTokenLog`, and uses it by default.
+//! - `platform-verifier`: Enabled by default. Provides `ClientConfig::with_platform_verifier()` convenience method.
+//! - `async-io`: TODO: What does this do? Seems to be related to smol?
+//! - `futures-io`: Enables `futures::io::{AsyncRead, AsyncWrite}` support for streams.
+//!
+//! ### Logging features
+//!
+//! This will cause a dependency on the `log` crate and will cause logs to be emitted at
+//! various log levels, for code in quinn or its dependencies.
+//!
+//! - `log`: Globally enables and disables logs. TODO: Is this true?
+//! - `qlog`: Enables logging in the quinn crates.
+//! - `rustls-log`: Enables logging in the rustls create.
+//! - `lock_tracking`: Enables logging of mutex locks in quinn. TODO: Does this enable lock tracking in deps?
+//!
+//! ### Crypto features
+//!
+//! The three supported cyrpto backends are rustls+ring, rustls+aws-lc-rs and rustls+aws-lc-rs-fips.
```

Comment:
Nit: "cyrpto" -> "crypto". See above for non-rustls crypto backends. I don't think this should go into detail about what happens if you have mismatched defaults, though it's probably good to point out that there is a mismatch.